### DISCUSSION TITLE
Failed to generate mol from InchI obtained from NIST database

### DISCUSSION THREAD

**Discussion by kharisecario** [rdkit/rdkit#7945] [Q&A] [answered]
hello, I would like to ask for help in generating mol from inchi obtained from NIST database.
The molecule in question is [Tetraheptylammonium bromide](https://webbook.nist.gov/cgi/cbook.cgi?InChI=1S%2FC28H60BrN%2Fc1-5-9-13-17-21-25-30%2829%2C26-22-18-14-10-6-2%2C27-23-19-15-11-7-3%2928-24-20-16-12-8-4%2Fh5-28H2%2C1-4H3&Units=SI).
with InChI=1S/C28H60BrN/c1-5-9-13-17-21-25-30(29,26-22-18-14-10-6-2,27-23-19-15-11-7-3)28-24-20-16-12-8-4/h5-28H2,1-4H3
When I use rdkit to generate mol from inchi, it give me an error:
```
Explicit valence for atom # 29 N, 5, is greater than permitted
ERROR: Explicit valence for atom # 29 N, 5, is greater than permitted
```
Any suggestion how to solve this?

**Comment by bertiewooster** [comment, marked as answer]
Try `MolFromInchi(inchi, sanitize=False)`. This will turn off sanitization and make the RDKit more lenient about making the input into a molecule.

**Comment by bertiewooster** [comment]
`sanitize=False` works:
```
inchi = "InChI=1S/C28H60BrN/c1-5-9-13-17-21-25-30(29,26-22-18-14-10-6-2,27-23-19-15-11-7-3)28-24-20-16-12-8-4/h5-28H2,1-4H3"
Chem.MolFromInchi(inchi)
```

```
[20:46:54] Explicit valence for atom # 29 N, 5, is greater than permitted
[20:46:54] ERROR: Explicit valence for atom # 29 N, 5, is greater than permitted
```

`Chem.MolFromInchi(inchi, sanitize=False)
`

![inchi](https://github.com/user-attachments/assets/c0635d19-87e8-46c3-9488-c84ad713d3e1)
